src/WebJobs.Script/FileProvisioning/PowerShell/PowerShellFileProvisioner.cs

```diff
@@ -30,22 +44,125 @@ public Task ProvisionFiles(string scriptRootPath)
+                string requirementsContent = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
+                string guidance = null;
+
+                try
+                {
+                    string majorVersion = GetLatestAzModuleMajorVersion();
+
+                    requirementsContent = requirementsContent.Replace("# 'Az'", "'Az'");
+                    requirementsContent = requirementsContent.Replace("MAJOR_VERSION", majorVersion);
+                }
+                catch
+                {
+                    guidance = "Uncomment the next line and replace the MAJOR_VERSION, e.g., 'Az' = '2.*'";
+                    _logger.LogWarning("Failed to get Az module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.");
+                }
+
+                requirementsContent = requirementsContent.Replace("[GUIDANCE]", guidance ?? string.Empty);
+                File.WriteAllText(requirementsFilePath, requirementsContent);
+
+                _logger.LogInformation($"{RequirementsPsd1FileName} created sucessfully.");
             }
         }
 
         private void AddProfileFile(string scriptRootPath)
         {
-            string profileFilePath = Path.Combine(scriptRootPath, "profile.ps1");
+            _logger.LogInformation($"Creating {ProfilePs1FileName}.");
+
+            string profileFilePath = Path.Combine(scriptRootPath, ProfilePs1FileName);
+
             if (!File.Exists(profileFilePath))
             {
                 string content = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.profile.ps1");
                 File.WriteAllText(profileFilePath, content);
             }
+
+            _logger.LogInformation($"{ProfilePs1FileName} created sucessfully.");
+        }
+
+        protected virtual string GetLatestAzModuleMajorVersion()
+        {
+            Uri address = new Uri($"{PowerShellGalleryFindPackagesByIdUri}'{AzModuleName}'");
+            string latestMajorVersion = null;
+            Stream stream = null;
+
+            var retryCount = 3;
+            while (true)
+            {
+                using (var client = new HttpClient())
+                {
+                    try
+                    {
+                        var response = client.GetAsync(address).Result;
+
+                        // Throw if not a successful request
+                        response.EnsureSuccessStatusCode();
+
+                        stream = response.Content.ReadAsStreamAsync().Result;
+                        break;
+                    }
+                    catch (Exception)
+                    {
+                        if (retryCount <= 0)
+                        {
+                            throw;
+                        }
+
+                        retryCount--;
+                    }
+                }
+            }
+
+            if (stream != null)
```

Comment:
To make this code modular and for testability purpose please consider moving the lines in a separate method that returns major version based upon the input Stream

Reply:
Agreed. Please split this method to make it easier to test

Reply:
Will do. Thanks.